udata_front/frontend/helpers.py

```diff
@@ -342,18 +342,15 @@ def i18n_alternate_links():
         LINK_PATTERN = (
             '<link rel="alternate" href="{url}" hreflang="{lang}" />')
         links = []
-        current_lang = get_current_locale().language
-
         params = {}
         if request.args:
             params.update(request.args)
         if request.view_args:
             params.update(request.view_args)
 
         for lang in current_app.config['LANGUAGES']:
-            if lang != current_lang:
-                url = url_for(request.endpoint, lang_code=lang, **params)
-                links.append(LINK_PATTERN.format(url=url, lang=lang))
+            url = url_for(request.endpoint, lang_code=lang, **params, _external=True)
```

Comment:
The `test_i18n_alternate_links` test should be updated to reflect the `_external=True` option + the current page alternate.